hack/cloud-image-uploader/azure.go

```diff
@@ -238,13 +265,13 @@ uploadLoop:
 func (azu *AzureUploader) createAzureDisk(ctx context.Context, armTemplate []byte, arch string) error {
 	diskParameters := map[string]interface{}{
 		"disk_name": map[string]string{
-			"value": "talos-" + arch + "-" + azu.Options.Tag,
+			"value": fmt.Sprintf("talos-%s-%s%s", arch, azu.Options.AzureCoreTag, azu.Options.AzurePreRelease),
```

Comment:
Append `-prerelease` to the core version for the disk image if it is a prerelease to keep the source consistent if overwritten during tests. If it is not a pre-release, this value is set to an empty string above.